Source/WebKit/NetworkProcess/storage/SQLiteStorageArea.cpp

```diff
@@ -165,6 +165,10 @@ bool SQLiteStorageArea::prepareDatabase(ShouldCreateIfNotExists shouldCreateIfNo
     auto openResult  = m_database->open(m_path);
     if (!openResult && handleDatabaseCorruptionIfNeeded(m_database->lastError())) {
         databaseExists = false;
+        if (shouldCreateIfNotExists == ShouldCreateIfNotExists::No)
+            return true;
```

Comment:
Are you sure that in case of corrupted database file and delete it we don't want to recreate empty database?

Reply:
If it's `ShouldCreateIfNotExists::No`, it means the caller does not need to create file or open connection  if database does not exist (for example, the caller wants to perform read operation, and if file is empty then there is nothing to read). 
In the case of corrupted database, after deletion, it's the same as the file does not exist.